t/t2260-job-list.t

```diff
@@ -723,6 +723,33 @@ test_expect_success 'flux job list of all jobs that ran on node[1-2] after certa
 	test $(cat constraint_hostlist9.out | wc -l) -eq ${numlines}
 '
 
+test_expect_success 'flux job list by rank (0-3)' '
+	constraint="{ and: [ {ranks:[\"0-3\"]} ] }" &&
+	$jq -j -c -n  "{max_entries:1000, attrs:[], constraint:${constraint}}" \
+	  | $RPC job-list.list | $jq .jobs | $jq -c '.[]' | $jq .id > constraint_hostlist1.out &&
+	numlines=$(cat completed.ids running.ids failed.ids timeout.ids | wc -l) &&
+	test $(cat constraint_hostlist1.out | wc -l) -eq ${numlines}
+'
+test_expect_success 'flux job list by rank (3)' '
+	constraint="{ and: [ {ranks:[\"3\"]} ] }" &&
+	$jq -j -c -n  \
+	  "{max_entries:1000, attrs:[\"ranks\"], constraint:${constraint}}" \
+	  | $RPC job-list.list | $jq -r .jobs[].ranks | uniq > jobs.ranks &&
+	test $(cat jobs.ranks) -eq 3
```

Comment:
sort stinks we have to hard code it.  could it be count of completed + running jobs divided by 4, or something like that?

Reply:
It may not be clear, but this is testing if all returned ranks were literally `3` (not a count of 3). That is, we queried for rank 3 and got only rank == 3 for all entries.

Reply:
yeah, that's obvious now ... i was mentally thinking of all the "get the right number of line" tests we have in the same area